File: src/Modules/Administration/ErpClink.Modules.Administration.Application/Users/Models/UserModels.cs
namespace ErpClink.Modules.Administration.Application.Users.Models;

public sealed record CreateUserRequest(
    string Email,
    string Password,
    string FullName,
    string? PhoneNumber,
    IReadOnlyList<string>? Roles);

public sealed record UpdateUserRequest(
    string FullName,
    string? PhoneNumber,
    string? Email);

public sealed record SetUserActiveRequest(bool IsActive);

public sealed record AssignRoleRequest(string RoleName);

public sealed record UserDto(
    string Id,
    string Email,
    string? UserName,
    string FullName,
    string? PhoneNumber,
    bool IsActive,
    DateTime CreatedAtUtc,
    DateTime? LastLoginAtUtc,
    IReadOnlyList<string> Roles);

File: frontend/erpclink-web/src/app/features/administration/models/user.models.ts
export interface CreateUserRequest {
  email: string;
  password: string;
  fullName: string;
  phoneNumber?: string | null;
  roles?: string[] | null;
}

export interface UpdateUserRequest {
  fullName: string;
  phoneNumber?: string | null;
  email?: string | null;
}

export interface UserDto {
  id: string;
  email: string;
  userName: string | null;
  fullName: string;
  phoneNumber: string | null;
  isActive: boolean;
  createdAtUtc: string;
  lastLoginAtUtc: string | null;
  roles: string[];
}
